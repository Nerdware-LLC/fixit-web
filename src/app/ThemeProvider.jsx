import React from "react";
import { ThemeProvider as MTP, createTheme } from "@mui/material/styles";
import CssBaseline from "@mui/material/CssBaseline";
import { any } from "../types";

export const ThemeProvider = ({ children }) => {
  const theme = createTheme(THEME);

  return (
    <MTP theme={theme}>
      <CssBaseline />
      {children}
    </MTP>
  );
};

const THEME = {
  description: "dark",
  palette: {
    mode: "dark",
    primary: {
      light: "#f89a35",
      main: "#f78103", // Fixit orange
      dark: "#ac5a02"
    },
    secondary: {
      light: "#5381ff",
      main: "#2962ff", // blue
      dark: "#1c44b2"
    },
    success: {
      light: "#33d373",
      main: "#00c851",
      dark: "#008c38"
    },
    error: {
      main: "rgba(255, 115, 115, 0.87)"
    },
    info: {
      main: "#ff80ab" // AKA NOTIFICATION
    },
    text: {
      primary: "rgba(255, 255, 255, 0.87)",
      disabled: "rgba(255, 255, 255, 0.38)",
      hint: "rgba(255, 255, 255, 0.54)", // AKA PLACEHOLDER
      labels: "#f7a44a" // desaturated Fixit orange (CUSTOM KEY)
    },
    background: {
      paper: "#353535", // AKA SURFACE
      default: "#121212",
      backdrop: "rgba(0, 0, 0, 0.5)" // (CUSTOM KEY)
    }
  }
};

ThemeProvider.propTypes = {
  children: any.isRequired
};
