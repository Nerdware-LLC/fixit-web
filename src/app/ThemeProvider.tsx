import React from "react";
import MuiThemeProvider from "@mui/material/styles/ThemeProvider";
import CssBaseline from "@mui/material/CssBaseline";
import { createTheme, type PaletteOptions } from "@mui/material/styles";
import { themeStore } from ".";
import { any } from "../types";

export const ThemeProvider = ({ children }: { children: React.ReactNode }) => {
  const theme = themeStore.useSubToStore();

  return (
    <MuiThemeProvider theme={THEMES[theme]}>
      <CssBaseline />
      {children}
    </MuiThemeProvider>
  );
};

/**
 * `ThemeProvider` merges this theme object with the @mui/material
 * default theme:
 *
 * https://mui.com/material-ui/customization/default-theme/
 */
const THEMES = Object.fromEntries(
  ["DARK", "LIGHT"].map((themeModeName) => [
    themeModeName,
    createTheme({
      palette: {
        mode: themeModeName.toLowerCase() as PaletteOptions["mode"],
        primary: {
          light: "#f89a35",
          main: "#f78103", // Fixit orange
          dark: "#ac5a02"
        },
        secondary: {
          light: themeModeName === "DARK" ? "#00FFDC" : "#625AFA",
          main: themeModeName === "DARK" ? "#00FFDC" : "#625AFA",
          dark: themeModeName === "DARK" ? "#00FFDC" : "#625AFA"
        },
        success: {
          light: "#33d373",
          main: "#00c851",
          dark: "#008c38"
        },
        error: {
          main: "rgba(255, 115, 115, 0.87)"
        },
        info: {
          main: "#5796D6" // AKA NOTIFICATION
        },
        text: {
          primary: themeModeName === "DARK" ? "rgba(255, 255, 255, 0.87)" : "rgba(0, 0, 0, 0.87)",
          secondary: themeModeName === "DARK" ? "#00FFDC" : "#625AFA",
          disabled: "rgba(255, 255, 255, 0.38)",
          hint: "rgba(255, 255, 255, 0.54)", // AKA PLACEHOLDER
          labels: "#f7a44a" // desaturated Fixit orange (CUSTOM KEY)
        } as any, // <-- Custom theme properties are fine.
        background: {
          paper: themeModeName === "DARK" ? "#353535" : "#cccccc", // AKA SURFACE
          default: themeModeName === "DARK" ? "#121212" : "#f6f6f6",
          backdrop: "rgba(0, 0, 0, 0.5)" // (CUSTOM KEY)
        } as any // <-- Custom theme properties are fine.
      },
      typography: {
        fontFamily: '"Roboto", sans-serif'
      }
    })
  ])
);

ThemeProvider.propTypes = {
  children: any.isRequired
};
