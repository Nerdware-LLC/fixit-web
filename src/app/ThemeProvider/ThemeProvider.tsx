import CssBaseline from "@mui/material/CssBaseline";
import MuiThemeProvider from "@mui/material/styles/ThemeProvider";
import { usePageLayoutContext } from "@app/PageLayoutContext/usePageLayoutContext";
import { themeStore } from "@app/apolloCache/reactiveVars/themeStore";
import { THEMES } from "./themes";
import type { Theme } from "@mui/material/styles";

export const ThemeProvider = ({ children }: ThemeProviderProps) => {
  const themeName = themeStore.useSubToStore();
  const { isMobileUserAgent, isMobilePageLayout } = usePageLayoutContext();

  const theme: Theme = {
    ...THEMES[themeName || "DARK"], // A fallback themeName for added safety (will crash if not set)
    variables: {
      isMobileUserAgent,
      isMobilePageLayout,
    },
  };

  return (
    <MuiThemeProvider theme={theme}>
      <CssBaseline />
      {children}
    </MuiThemeProvider>
  );
};

export type ThemeProviderProps = { children: React.ReactNode };
