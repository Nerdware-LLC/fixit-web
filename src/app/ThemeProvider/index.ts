export { ThemeProvider } from "./ThemeProvider";
export * from "./themes";
