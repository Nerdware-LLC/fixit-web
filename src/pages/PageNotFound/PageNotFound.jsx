import React from "react";
import styled from "@emotion/styled";
import { useWebViewContext, Title, Text, Button } from "../../components";

export const PageNotFound = () => {
  const { webViewPostMessage } = useWebViewContext();

  const exitWebViewWithError = () => {
    webViewPostMessage({ error: "INVALID_ROUTE" });
  };

  return (
    <StyledPageNotFoundLayout>
      <Title>Whoops - this page doesn&apos;t seem to exist!</Title>
      <Text>
        <span role={"img"} aria-label={"x emoji"}>
          ‚ùå
        </span>{" "}
        404 - File or directory not found.
      </Text>
      <Text>
        The resource you are looking for might have been removed, had its name
        changed, or is temporarily unavailable.
      </Text>
      <Button
        label={"Take Me Back to Civilization!"}
        onClick={exitWebViewWithError}
        style={{ alignSelf: "center", fontWeight: "bold" }}
        variant={"contained"}
        color={"primary"}
        size={"large"}
      />
    </StyledPageNotFoundLayout>
  );
};

const StyledPageNotFoundLayout = styled.div`
  box-sizing: border-box;
  height: 100%;
  text-align: left;
  display: flex;
  flex-direction: column;
  padding: 30vh 15vw;
  align-items: center;
  justify-content: space-around;
`;
