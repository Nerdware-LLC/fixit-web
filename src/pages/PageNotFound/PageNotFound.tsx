import React from "react";
import { useNavigate } from "react-router-dom";
import { styled } from "@mui/material/styles";
import Button from "@mui/material/Button";
import Divider, { dividerClasses } from "@mui/material/Divider";
import Text, { typographyClasses } from "@mui/material/Typography";
import { DragonIcon } from "@components/Icons/DragonIcon";
import { Anchor } from "@components/Navigation/Anchor";
import { Link } from "@components/Navigation/Link";

export const PageNotFound = () => {
  const nav = useNavigate();

  return (
    <StyledDiv>
      <div className={classNames.headerContainer}>
        <Text variant="h2">
          <span className={classNames.text404}>404</span>Whoops - this page doesn&apos;t seem to
          exist!
        </Text>
      </div>
      <div className={classNames.bodyContainer}>
        <div>
          <Text>Looking for something in particular? Hopefully one of these links will help:</Text>
          <ul>
            {[
              { to: "/", label: "Home" },
              { to: "/pricing", label: "Pricing" },
              { to: "/privacy", label: "Privacy Policy" },
              { to: "/tos", label: "Terms of Service" },
            ].map(({ to, label }, index) => (
              <React.Fragment key={label}>
                {index !== 0 && <Divider orientation="vertical" flexItem />}
                <li key={to}>
                  <Link to={to}>{label}</Link>
                </li>
              </React.Fragment>
            ))}
          </ul>
        </div>
        <Text>
          Sure,{` `}
          <Anchor href="https://www.sparknotes.com/lit/fellowship/quotes/page/4/#:~:text=Quote%204,whether%20Strider%20is%20indeed%20Aragorn.">
            "Not all those who wander are lost"
          </Anchor>
          {`  `}
          and all that, but <i>you</i> - my friend - have wandered right off the map and straight
          into <em>"There be dragons" </em> territory! <DragonIcon />
        </Text>
      </div>
      <div className={classNames.buttonContainer}>
        <Button onClick={() => nav("/")} size="large">
          Take me home!
        </Button>
      </div>
    </StyledDiv>
  );
};

const classNames = {
  text404: "page-not-found-text-404",
  headerContainer: "page-not-found-header-container",
  bodyContainer: "page-not-found-body-container",
  buttonContainer: "page-not-found-button-container",
};

const StyledDiv = styled("div")(({ theme: { breakpoints, variables } }) => ({
  height: "100%",
  padding: "3rem 2rem",
  display: "flex",
  flexDirection: "column",
  alignItems: "center",
  justifyContent: "flex-start",
  gap: "2rem",
  textAlign: "center",

  [breakpoints.up("md")]: {
    justifyContent: "center",
    padding: "4rem",
    gap: "4rem",
  },

  // ALL CHILD DIV-CONTAINERS (low-specificity)
  "& > div": {
    display: "flex",
    alignItems: "center",
    justifyContent: "center",
    maxWidth: "66rem",
  },

  // HEADER
  [`& > div.${classNames.headerContainer} > h2.${typographyClasses.root}`]: {
    ...(variables.isMobilePageLayout
      ? { fontSize: "2rem", lineHeight: "2.25rem" }
      : { fontSize: "3.25rem", lineHeight: "3.25rem" }),
    // "404" text
    [`& .${classNames.text404}`]: {
      opacity: 0.3,
      marginRight: "1rem",
    },
  },

  // BODY-CONTAINER
  [`& > div.${classNames.bodyContainer}`]: {
    flexDirection: "column",
    gap: "inherit",
    ...(variables.isMobilePageLayout
      ? { fontSize: "1rem", lineHeight: "1.5rem" }
      : { fontSize: "1.5rem", lineHeight: "2rem" }),

    [`& .${typographyClasses.root}`]: {
      fontSize: "inherit",
      lineHeight: "inherit",
    },

    "& > div > ul": {
      margin: "0.5rem 0 0 0",
      padding: 0,
      display: "flex",
      alignItems: "center",
      justifyContent: "center",
      flexDirection: variables.isMobilePageLayout ? "column" : "row",
      "& > li": {
        listStyleType: "none",
        margin: "0 1rem",
        "& > a": {
          textDecoration: "underline",
        },
      },
      [`& > .${dividerClasses.root}`]: {
        ...(variables.isMobilePageLayout && { display: "none" }),
      },
    },

    [`& > .${typographyClasses.root}`]: {
      whiteSpace: "pre-wrap",

      "& a,em": {
        whiteSpace: "nowrap",
        fontStyle: "italic",
      },
      "& svg": {
        ...(variables.isMobilePageLayout
          ? {
              height: "1rem",
              marginLeft: "0.15rem",
              transform: "translateY(2px)",
            }
          : {
              height: "1.5rem",
              marginLeft: "0.35rem",
              transform: "translateY(3px)",
            }),
      },
    },
  },

  // BUTTON
  [`& > div.${classNames.buttonContainer} > button`]: {
    marginTop: "0.5rem",
    fontStyle: "italic",
  },
}));
