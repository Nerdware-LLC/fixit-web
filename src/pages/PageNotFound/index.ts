export { PageNotFound as default } from "./PageNotFound";
// export as default to enable code-splitting via lazy import.
