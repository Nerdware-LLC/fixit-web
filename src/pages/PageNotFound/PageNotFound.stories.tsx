import { withRootAppLayoutDecorator } from "@/../.storybook/decorators";
import { PageNotFound } from "./PageNotFound.jsx";
import type { Meta, StoryObj } from "@storybook/react";

const meta = {
  title: "Pages/PageNotFound",
  component: PageNotFound,
  decorators: [withRootAppLayoutDecorator],
  parameters: {
    layout: "fullscreen",
  },
} satisfies Meta<typeof PageNotFound>;

export default meta;

///////////////////////////////////////////////////////////
// STORIES

type Story = StoryObj<typeof meta>;

export const BasicDemo = {} satisfies Story;
