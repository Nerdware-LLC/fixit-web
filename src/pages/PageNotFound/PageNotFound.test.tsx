import renderer from "react-test-renderer";
import { MemoryRouter, Routes, Route } from "react-router-dom";
import { PageNotFound } from "./PageNotFound";
import { ThemeProvider } from "@app/ThemeProvider";

it("renders correctly", () => {
  expect(
    renderer
      .create(
        <ThemeProvider>
          <MemoryRouter initialEntries={["/test-page-not-found"]}>
            <Routes>
              <Route path="*" element={<PageNotFound />} />
            </Routes>
          </MemoryRouter>
        </ThemeProvider>
      )
      .toJSON()
  ).toMatchSnapshot();
});
