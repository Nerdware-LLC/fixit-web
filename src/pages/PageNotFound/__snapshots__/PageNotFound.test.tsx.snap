// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders correctly 1`] = `
<div
  className="css-172n9ff"
>
  <div
    className="page-not-found-header-container"
  >
    <h2
      className="MuiTypography-root MuiTypography-h2 css-tptap9-MuiTypography-root"
    >
      <span
        className="page-not-found-text-404"
      >
        404
      </span>
      Whoops - this page doesn't seem to exist!
    </h2>
  </div>
  <div
    className="page-not-found-body-container"
  >
    <div>
      <p
        className="MuiTypography-root MuiTypography-body1 css-1e7uc26-MuiTypography-root"
      >
        Looking for something in particular? Hopefully one of these links will help:
      </p>
      <ul>
        <li>
          <a
            className="css-cpk69e"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <hr
          className="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1kpnj7b-MuiDivider-root"
        />
        <li>
          <a
            className="css-cpk69e"
            href="/pricing"
            onClick={[Function]}
          >
            Pricing
          </a>
        </li>
        <hr
          className="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1kpnj7b-MuiDivider-root"
        />
        <li>
          <a
            className="css-cpk69e"
            href="/privacy"
            onClick={[Function]}
          >
            Privacy Policy
          </a>
        </li>
        <hr
          className="MuiDivider-root MuiDivider-fullWidth MuiDivider-vertical MuiDivider-flexItem css-1kpnj7b-MuiDivider-root"
        />
        <li>
          <a
            className="css-cpk69e"
            href="/tos"
            onClick={[Function]}
          >
            Terms of Service
          </a>
        </li>
      </ul>
    </div>
    <p
      className="MuiTypography-root MuiTypography-body1 css-1e7uc26-MuiTypography-root"
    >
      Sure,
       
      <a
        className="css-cpk69e"
        href="https://www.sparknotes.com/lit/fellowship/quotes/page/4/#:~:text=Quote%204,whether%20Strider%20is%20indeed%20Aragorn."
        rel="noreferrer"
        target="_blank"
      >
        "Not all those who wander are lost"
      </a>
        
      and all that, but 
      <i>
        you
      </i>
       - my friend - have wandered right off the map and straight into 
      <em>
        "There be dragons" 
      </em>
       territory! 
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
        focusable="false"
        height="1em"
        viewBox="0 0 640 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M352 124.5l-51.9-13c-6.5-1.6-11.3-7.1-12-13.8s2.8-13.1 8.7-16.1l40.8-20.4L294.4 28.8c-5.5-4.1-7.8-11.3-5.6-17.9S297.1 0 304 0H416h32 16c30.2 0 58.7 14.2 76.8 38.4l57.6 76.8c6.2 8.3 9.6 18.4 9.6 28.8c0 26.5-21.5 48-48 48H538.5c-17 0-33.3-6.7-45.3-18.7L480 160H448v21.5c0 24.8 12.8 47.9 33.8 61.1l106.6 66.6c32.1 20.1 51.6 55.2 51.6 93.1C640 462.9 590.9 512 530.2 512H496 432 32.3c-3.3 0-6.6-.4-9.6-1.4C13.5 507.8 6 501 2.4 492.1C1 488.7 .2 485.2 0 481.4c-.2-3.7 .3-7.3 1.3-10.7c2.8-9.2 9.6-16.7 18.6-20.4c3-1.2 6.2-2 9.5-2.2L433.3 412c8.3-.7 14.7-7.7 14.7-16.1c0-4.3-1.7-8.4-4.7-11.4l-44.4-44.4c-30-30-46.9-70.7-46.9-113.1V181.5v-57zM512 72.3c0-.1 0-.2 0-.3s0-.2 0-.3v.6zm-1.3 7.4L464.3 68.1c-.2 1.3-.3 2.6-.3 3.9c0 13.3 10.7 24 24 24c10.6 0 19.5-6.8 22.7-16.3zM130.9 116.5c16.3-14.5 40.4-16.2 58.5-4.1l130.6 87V227c0 32.8 8.4 64.8 24 93H112c-6.7 0-12.7-4.2-15-10.4s-.5-13.3 4.6-17.7L171 232.3 18.4 255.8c-7 1.1-13.9-2.6-16.9-9s-1.5-14.1 3.8-18.8L130.9 116.5z"
        />
      </svg>
    </p>
  </div>
  <div
    className="page-not-found-button-container"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge css-1npnc9i-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Take me home!
    </button>
  </div>
</div>
`;
